js/ng.js

```diff
@@ -77,7 +77,7 @@ addEventListener('deer-form-rendered', event => {
             prefillThemesArea(annotationData["themes"], event.target)
             prefillText(annotationData["text"], event.target)
             if(event.detail.targetChapter && !event.detail.section) {
-                document.querySelector('[deer-key="canonicalReference"]').value = `Matthew ${event.detail.targetChapter.value}:${event.detail.targetVerse.value}`
+                document.querySelector('[deer-key="canonicalReference"]').value = `Matthew ${event.detail.targetChapter.value || ''}${event.detail.targetVerse.value ? `:${event.detail.targetVerse.value}` : ''}`
```

Comment:
From testing, I no longer see any `undefined` appearances!

Looks like the right place and the right logic.  Do both the checks of `targetVerse` and `targetChapter` in the same syntax style. 

```javascript

 document.querySelector('[deer-key="canonicalReference"]').value = `Matthew ${event.detail.targetChapter.value ?  event.detail.targetChapter.value : ''}${event.detail.targetVerse.value ? `:${event.detail.targetVerse.value}` : ''}`
 
```

Reply:
Oh nevermind!  That extra `:` you add in is why you didn't do them both like ` || ''`